src/main/java/com/jfrog/ide/idea/configuration/XrayServerConfigImpl.java

```diff
@@ -197,10 +199,30 @@ void setPassword(String password) {
     }
 
     void setCredentials(Credentials credentials) {
+        if (credentials == null) {
+            return;
+        }
         setUsername(credentials.getUserName());
         setPassword(credentials.getPasswordAsString());
     }
 
+    /**
+     * Initialize connection details from environment variables.
+     * Minimal required details are url and username.
+     */
+    public void initConnectionDetailsFromEnv() {
+        String urlEnv = System.getenv(URL_ENV);
```

Comment:
This approach should work on 100% of the cases. See explanation above [getEnvironmentMap()](https://github.com/JetBrains/intellij-community/blob/webstorm/193.5233.13/platform/util/src/com/intellij/util/EnvironmentUtil.java#L128).
```suggestion
        String urlEnv = EnvironmentUtil.getValue(URL_ENV);
```